File: exercises/java/src/main/java/bootcamp/AnimalCard.java
package bootcamp;

public class AnimalCard {

    private final Animal animal;

    AnimalCard(Animal animal) {
        this.animal = animal;
    }

    boolean snap( AnimalCard otherCard ){
        return this.animal.equals(otherCard.animal);
    }

    @Override
    public String toString() {
        return animal.toString();
    }
}

File: exercises/swift/SnapGameOOP/SnapGameOOP/GameLogic/Cards/Animal/AnimalCard.swift

//
//  AnimalCard.swift
//  SnapGameOOP
//
//  Created by Andrew Harris on 19/11/2019.
//

import Foundation

class AnimalCard {
    
    private let animal: Animal
    
    init(animal: Animal) {
        self.animal = animal
    }

}

extension AnimalCard: Card {
    func snap(otherCard: Card) -> Bool {
        return otherCard is AnimalCard && self.getValue() == otherCard.getValue()
    }
    
    func getValue() -> String {
        self.animal.value()
    }
}
